### PR TITLE
Subsets Generation

### PR DESCRIPTION
**Pull request by miretteamin**
## Related Issue
- Issue #1305 

## Proposed Changes
- I added the implementation of the algorithm.
- I updated the Readme file accordingly.

## Additional Info
- Readme has all the information and also I included comments in the code.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

I implemented subset generation algorithm using bit masking.
This algorithm prints all possible subsets from a given array with any data type.

Bit masking is faster than using ordinary operators and consumes less memory.

I included 2 examples, the first one was testing an integer array and I took this test from Subset problem on Leetcode, and the second test was testing a character array.

I ran my tests on Visual Studio Community 2019, CPP compiler.
The 2 tests are included in the screenshots.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Code Screenshot | Tests Screenshot |
| image  | image  |

### CODE REVIEW

File: B-BitManipulation/SubsetsGeneration.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	vector<vector<T>>res((1 << s));
+	for (int mask = 0; mask < (1 << s); mask++)
+	{
+		for (int i = 0; i < s; i++) {
+			if ((mask >> i) & 1)// is the i-th bit of this mask 1?
+				res[mask].push_back(arr[i]);
+		}
+	}
+	return res;
+}
```
Comment (Mrjoy832) [inline B-BitManipulation/SubsetsGeneration.cpp:23]: add more comments on code

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
LGTM🌟

File: B-BitManipulation/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Input 1<br>
+![image](https://user-images.githubusercontent.com/29145628/166822148-f5507a95-2a05-4b66-b118-8d610bf433d3.png)<br>
+Output 1<br>
+![image](https://user-images.githubusercontent.com/29145628/166822236-124ac15b-6722-4a94-a877-0bc0abe36691.png)<br>
+
+> Test 2<br>
+Input 2<br>
+![image](https://user-images.githubusercontent.com/29145628/166822297-fbc277b6-1bbe-4b0d-85a7-0b961e6299f3.png)<br>
+Output 2<br>
+![image](https://user-images.githubusercontent.com/29145628/166822363-aa8d8e4f-b358-4e35-99b3-ea5ad492597b.png)
```
Comment (Lakhankumawat) [inline B-BitManipulation/README.md:48]: Remove this section we don't require outputs anymore

File: B-BitManipulation/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- [Subsets Generation using Bit Masking](#subsets-generation-using-bit-masking)
+    - [Algorithm](#algorithm)
+    - [Time Complexity](#time-complexity)
+    - [Space Complexity](#space-complexity)
+    - [Tests](#output)
+
+
+
+## Subsets Generation using Bit Masking
+Algorithm that uses bit masking to generate all possible subsets from a given array with a chosen data type.<br>
```
Comment (Lakhankumawat) [inline B-BitManipulation/README.md:11]: Add a working image of algorithm 

You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Review by Lakhankumawat — APPROVED** [review]
## Looks Awesome ✅

**Comment by Lakhankumawat** [comment]
Continuous Good Contribution , mention this pr number in your next pr and your next pr will have higher changes of getting higher level if found eligible .

**Comment by miretteamin** [comment]
> Continuous Good Contribution , mention this pr number in your next pr and your next pr will have higher changes of getting higher level if found eligible .

Thank you!!!!

**Pull request merged**